### ISSUE TITLE
Function Not Found - basic-rust-lambda-dev-hello

### ISSUE THREAD

**Issue by JimLynchCodes** [softprops/serverless-rust#62]
## 🐛 Bug description
Describe your issue in detail.

When I scaffold a fresh project with `npx serverless install \
  --url https://github.com/softprops/serverless-aws-rust-http \
  --name my-new-api` then run `npm i` and then try to invoke locally:

```
npx serverless invoke -f hello -d "$(cat payload.json)"
```

It gives me the error, "Function not found".

Serverless Error ---------------------------------------
 ```
  Function not found: arn:aws:lambda:us-east-1:954459734159:function:basic-rust-lambda-dev-hello
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     OS:                     darwin
     Node Version:           11.15.0
     Serverless Version:     1.47.0
```

![Screen Shot 2020-01-26 at 11 40 56 AM](https://user-images.githubusercontent.com/5354163/73138509-59a85280-4031-11ea-9ded-fa2ae9c97fbb.png)

#### 🤔 Expected Behavior

It should have found the function `hello`.

#### 👟 Steps to reproduce

(described above)

#### 🌍 Your environment

serverless version: 1.47.0

rust-plugin version: 0.3.6
 Node Version:           11.15.0
   OS:                     darwin

**Comment by JimLynchCodes** [comment]
Example project here - > https://github.com/JimLynchCodes/Basic-Rust-Lambda

Thanks! 🙏

**Comment by brainstorm** [comment]
What does `sls deploy` say?

**Comment by JimLynchCodes** [comment]
hmm not sure if I deployed it actually. 

I don't want to deploy it yet though- I just want to invoke it locally. Is there a recommended way to do that?

Thanks! ❤️

**Comment by brainstorm** [comment]
According to README.md, passing `local` keyword on the npx cmdline:

https://github.com/softprops/serverless-rust#invoke-your-lambdas-locally

You don't seem to be doing that AFAICT?

**Comment by JimLynchCodes** [comment]
Thank you @brainstorm. I am still having issues though.

1. I cloned the repo again
2. I ran `npm i`
3. I run the invoke script including `local`:
```
npx serverless invoke local -f hello -d "$(cat local-invoke-payload.json)"
```

But I am still getting errors about docker for some reason:
```
   │                                                                │
   │      New minor version of npm available! 6.12.1 → 6.13.7       │
   │   Changelog: https://github.com/npm/cli/releases/tag/v6.13.7   │
   │               Run npm install -g npm to update!                │
   │                                                                │
   ╰────────────────────────────────────────────────────────────────╯

basic-rust-lambda (master) ♘ npx serverless invoke local -f hello -d "$(cat local-invoke-payload.json)"
Serverless: Building native Rust hello func...
docker: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?.
See 'docker run --help'.
Serverless: Dockerized Rust build encountered an error: undefined 125.
 
  Error --------------------------------------------------
 
 
     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     OS:                     darwin
     Node Version:           12.13.1
     Serverless Version:     1.47.0
```

Maybe I need to update versions of something?

I am really interested in building lambda functions with Rust so I'm very interested in getting this working! Thanks! 🙏❤️

**Comment by brainstorm** [comment]
`docker: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?.`... your docker daemon doesn't seem to be running.

**Comment by JimLynchCodes** [comment]
Can you please elaborate?

What is the command to start the docker daemon running for this project?

If it is necessary, can we please add it to the README?

**Comment by JimLynchCodes** [comment]
@brainstorm Can you please elaborate?

What is the command to start the docker daemon running for this project?

If it is necessary, can we please add it to the README?

**Comment by JimLynchCodes** [comment]
@brainstorm ?

**Comment by MMuthukrishna** [comment]
@JimLynchCodes 
please install docker with `brew cask install docker` or any other package manager available for mac, and start docker by opening it.
You can start docker by opening it from Applications in Finder or from Spotlight search

this level of usage instructions cannot be added to README.

**Comment by JimLynchCodes** [comment]
🙃

**Comment by softprops** [comment]
Hi folks sorry for the delayed response. @JimLynchCodes I tried my best to catch and add useful logging on the case I think we were hitting. The docker daemon was not running. It is needed for the default build for this plugin but also by serverless itself for locally invoking functions on the aws provided runtime.


Of note I'm currently publishing a release for experimental local builds - https://github.com/softprops/serverless-rust#-experimental-local-builds you might want to try this out if docker is becoming a hassle.

Closing the this issue because both reported outcomes were expected behavior for invoking a remote lambda without deploying and invoking locally without a running docker daemon
